Sources/Client/Model/Message/MessageRead.swift

```diff
@@ -8,26 +8,31 @@
 
 import Foundation
 
-/// A message read state. User + last read date.
+/// A message read state. User + last read date + unread message count.
 public struct MessageRead: Decodable, Hashable {
     private enum CodingKeys: String, CodingKey {
         case user
         case lastReadDate = "last_read"
+        case unreadMessageCount = "unread_messages"
     }
     
     /// A user (see `User`).
     public let user: User
     /// A last read date by the user.
     public let lastReadDate: Date
+    /// Unread message count for the user.
+    public let unreadMessageCount: Int
     
     /// Init a message read.
     ///
     /// - Parameters:
     ///   - user: a user.
     ///   - lastReadDate: the last read date.
-    public init(user: User, lastReadDate: Date) {
+    ///   - unreadMessages: Unread message count
+    public init(user: User, lastReadDate: Date, unreadMessageCount: Int) {
```

Comment:
We need a default value for the backward compatibility.

Reply:
This is `public` because it's used in Core & UI, I don't think anyone needs to initialize this. Do you think we should add a default value?

Reply:
It's your call 🤷‍♂️